langserver/handlers/service.go

```diff
@@ -166,6 +166,11 @@ func (svc *service) Assigner() (jrpc2.Assigner, error) {
 			ctx = lsctx.WithRootModuleManager(ctx, svc.modMgr)
 			ctx = lsctx.WithRootModuleLoader(ctx, rmLoader)
 
+			version, ok := lsctx.LanguageServerVersion(svc.srvCtx)
```

Comment:
I don't get the cycle here?

Reply:
This is because the server starts with its own empty context, for better or worse:
https://github.com/creachadair/jrpc2/blob/3fb8081ec1bde4adba1b3a57dff458328a8ea769/server.go#L304

You could argue both sides - it's good for isolation - you don't get to inherit things by accident, but also it can make it more verbose 🤷‍♂️

Reply:
ah okay!